src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -491,7 +491,7 @@ export function registerTerminalActions() {
 				return;
 			}
 			c.service.setActiveInstance(instance);
-			return c.groupService.showPanel(true);
+			c.service.focusActiveInstance();
```

Comment:
I think this is the one we want which doesn't just focus but also reveals:

```suggestion
			focusActiveInstance(instance, c);
```